src/utils/validationSchema.js

```diff
@@ -0,0 +1,28 @@
+import Joi from 'joi';
+import errorMessage from './errormessage.js';
+
+const LoginSchema = Joi.object().keys({
+  email: Joi.string().email().required().messages(errorMessage('Email')),
+  password: Joi.string()
+    .pattern(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0123456789])(?=.*[@$!%*?&])[A-Za-z0123456789@$!%*?&]{8,}$/
+    )
+    .messages(errorMessage('Password')), // password has both numbers and letters and is btn 6 and 30
+});
+
+const SignUpSchema = Joi.object().keys({
+  username: Joi.string()
+    .min(6)
+    .max(30)
+    .required()
+    .messages(errorMessage('Username')),
+  email: Joi.string().email().required().messages(errorMessage('Email')),
+  password: Joi.string()
+    .required()
+    .pattern(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0123456789])(?=.*[@$!%*?&])[A-Za-z0123456789@$!%*?&]{8,}$/
+    )
+    .messages(errorMessage('Password')), // password has both numbers and letters and is btn 6 and 30
+});
+
+export { LoginSchema, SignUpSchema };
```

Comment:
it seams like you are changing all the files with no specific lines. did you copy and paste, or rebase?

Reply:
I rebased.